common/src/main/java/com/microsoft/identity/common/internal/cache/AbstractAccountCredentialCache.java

```diff
@@ -209,6 +213,11 @@
                 matches = matches && authScheme.equalsIgnoreCase(atType);
             }
 
+            if (mustMatchOnRequestedClaims && credential instanceof AccessTokenRecord) {
```

Comment:
(wishlist, not strictly reqd) Can we (similar to above with `mustMatchOnTarget`) log a message if the caller tries to filter on `requestedClaims` for an a credential type this can't apply to?

For instance, if the developer tries to match `target` to an IdToken, they will get the message: `Logger.verbose(TAG, "Query specified target-match, but no target to match.");`. As a "nice to have", a similar message could be useful for non-AT credential types

Reply:
Added in latest commit!